src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -674,9 +682,57 @@ def list(self):
             print_str += " {0} {1}".format(time_string, ccp)
             if kw_idx == 0:
                 print_str += " {0}".format(self._products_df_sorted["producttype"][idx])
+                print_str += " {0}".format(self._products_df_sorted["size"][idx])
 
             print(print_str)
 
+    def skip_existing(self, output, pattern_file):
+        prod_df_type = type(self._products_df_sorted)
+        # Check i skipping is possible/required
+        if prod_df_type != dict:
+            if self._products_df_sorted.empty:
+                return
+        elif not self._products_df_sorted or os.path.exists(output) == False:
+            return
+        # Check if ingestion date is returned by API
+        if "ingestiondate" not in self._products_df_sorted:
+            gs.warning(
+                _(
+                    "Ingestiondate not returned. Cannot filter previously downloaded scenes"
+                )
+            )
+            return
+        # Check for previously downloaded scenes
+        existing_files = glob(os.path.join(output, pattern_file))
+        if len(existing_files) <= 1:
+            return
+        # Filter by ingestion date
+        skiprows = []
+        for idx, display_id in enumerate(self._products_df_sorted["identifier"]):
+            existing_file = [sfile for sfile in existing_files if display_id in sfile]
+            if existing_file:
+                creation_time = datetime.fromtimestamp(
+                    os.path.getctime(existing_file[0])
+                )
+                if self._products_df_sorted["ingestiondate"][idx] <= creation_time:
+                    gs.verbose(
```

Comment:
I would suggest to change this to `gs.message()` so that it is always visible.

Reply:
Good point. Integrated in #609